File: Example/javaapp/src/main/java/com/stripe/example/javaapp/fragment/location/LocationListState.java
package com.stripe.example.javaapp.fragment.location;

import com.stripe.stripeterminal.external.models.Location;

import org.jetbrains.annotations.NotNull;

import java.util.Collections;
import java.util.List;

/**
 * Represents the current view state of the location list.
 */
final public class LocationListState {
    private List<Location> locations;
    private boolean isLoading;
    private boolean hasMore;

    public LocationListState(List<Location> locations, boolean isLoading, boolean hasMore) {
        this.locations = locations;
        this.isLoading = isLoading;
        this.hasMore = hasMore;
    }

    public LocationListState() {
        this(Collections.emptyList(), true, true);
    }

    public boolean hasMore() {
        return hasMore;
    }

    public boolean isLoading() {
        return isLoading;
    }

    public List<Location> getLocations() {
        return locations;
    }

    /**
     * @return Whether the header view should be displayed in the list.
     */
    public boolean headerVisible() {
        return !locations.isEmpty() || !isLoading();
    }
}

File: Example/kotlinapp/src/main/java/com/stripe/example/fragment/location/LocationListState.kt
package com.stripe.example.fragment.location

import com.stripe.stripeterminal.external.models.Location

/**
 * Represents the current view state of the location list.
 */
data class LocationListState(
    val locations: List<Location> = emptyList(),
    val isLoading: Boolean = true,
    val hasMore: Boolean = true,
) {
    /**
     * Whether the header view should be displayed in the list.
     */
    val headerVisible = locations.isNotEmpty() || !isLoading
}
